lib/foreman/cast.rb

```diff
@@ -0,0 +1,26 @@
+module Foreman
+  module Cast
+    def self.to_bool(value)
+      case value
+
+      when String
+        return true if value =~ (/\A(true|t|yes|y|on|1)\z/i)
+        return false if value.blank? || value =~ (/\A(false|f|no|n|off|0)\z/i)
```

Comment:
How is an empty object "effectively `nil`"?

I think if you are calling `to_bool()` on a value then you are expecting that value to be transformed to a boolean if possible (it as a casting method after all).  If you need to check whether value is nil then you wouldn't call `to_bool()` on it.

In your example `to_bool()` would return `nil` for the empty object, because it wouldn't match any of the known conversions, thus still providing a means of differentiating between the two.